packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -189,3 +189,92 @@ describe('findSpecs', () => {
     expect(specs).to.have.length(3)
   })
 })
+
+describe('getDefaultSpecFileName', () => {
+  context('dirname', () => {
+    it('returns pattern without change if it is do not a glob', () => {
+      const specPattern = 'cypress/e2e/foo.spec.ts'
+      const defaultFileName = getDefaultSpecFileName(specPattern)
+
+      expect(defaultFileName).to.eq(specPattern)
+    })
+
+    it('remove ** from glob if it is not in the beginning', () => {
+      const defaultFileName = getDefaultSpecFileName('cypress/**/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/foo.spec.ts')
+    })
+
+    it('replace ** for cypress if it starts with **', () => {
+      const defaultFileName = getDefaultSpecFileName('**/e2e/foo.spec.ts')
+
+      expect(defaultFileName).to.eq('cypress/e2e/foo.spec.ts')
+    })
+
+    it('replace ** for cypress if it starts with ** and omit extra **', () => {
```

Comment:
Can we add a test for something like `e2e/**/*.cy.{js,jsx,ts,tsx}`

Reply:
https://github.com/cypress-io/cypress/commit/5deb317f0cf72ae584ef385345a92c43e8bf80ef#diff-59bf8023d3f96e2d91bc479be5372ebde0317cf6040e5e514ceb276ac71c1342R334-R338